packages/abi-coder/src/encoding/strategies/getCoderV1.ts

```diff
@@ -126,21 +127,26 @@ export const getCoder: GetCoderFn = (
     return new VecCoder(itemCoder as Coder);
   }
 
-  const structMatch = structRegEx.exec(resolvedAbiType.type)?.groups;
+  const structMatch = structRegEx.test(resolvedAbiType.type);
   if (structMatch) {
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```

Comment:
Just as a suggestion to avoid the _no-non-null assertion_ rule 

```ts
function getFirstMatch(str: string, regex: RegExp): string | null {
  const match = str.match(regex);
  return match ? match[0] : null;
}

const structMatch = structRegEx.test(resolvedAbiType.type);
if (structMatch) {
  const name = getFirstMatch(resolvedAbiType.type, fullNameRegExMatch);
  if (name) {
    const coders = getCoders(components, { getCoder });
    return new StructCoder(name, coders);
  }
}
```

Reply:
For accessing array indexes conditionally, you can simply:

```ts
resolvedAbiType.type.match(fullNameRegExMatch)?.[0]
```

https://github.com/FuelLabs/fuels-ts/pull/2856/commits/5621ccfd433f58e73f9188fb03f05088223ca57b